Popup: describe tabFocusLoopEnabled (#8602)
DevExpress/devextreme-documentation: api-reference/10 UI Components/dxPopup/1 Configuration/tabFocusLoopEnabled.md

---
---
##### shortDescription

> Specifies whether focus loops within the popup content when the Tab key is pressed.

Reviewer: ```suggestion
Specifies whether focus stays within the popup content when the Tab key is pressed.
```

---

> When this option is `true` and the popup has input focus, the Tab key cycles through focusable elements in that popup. When focus reaches the last focusable element, the Tab key moves focus to the first focusable element. Shift+Tab combination reverses the focus order.

Reviewer: ```suggestion
When this option is `true` and the popup has focus, the Tab key cycles through focusable elements in the popup. After the last focusable element, focus returns to the first element. Press Shift+Tab to move focus in reverse order.
```

> When this option is `true`, pressing the Tab key when focus is within the popup cycles through the focusable elements in the popup. When focus reaches the last focusable element, pressing Tab again moves focus to the first focusable element. Pressing Shift+Tab reverses the focus order.

Reviewer 2: I would try to avoid an ING word as a subject. (And you can always says "the key does X" instead of "pressing the key does X"). 

And I don't want to repeat "within the popup / in the popup" in one sentence. 

```suggestion
When this option is `true` and the popup has input focus, the Tab key cycles through focusable elements in that popup. When focus reaches the last focusable element, the Tab key moves focus to the first focusable element. Shift+Tab combination reverses the focus order.
```

> When this option is `false` (default) and input focus is on the last focusable element, the Tab key moves focus outside the popup.

Reviewer: ```suggestion
When this option is `false` (default) and focus is on the last focusable element, the Tab key moves focus outside the popup.
```

> When this option is `false` (default), pressing the Tab key on the last focusable element moves focus outside the popup.

Reviewer 2: ```suggestion
When this option is `false` (default) and input focus is on the last focusable element, the Tab key moves focus outside the popup.
```

> This option has no effect when [shading](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#shading) is enabled. In this case, focus always loops within the popup content regardless of the `tabFocusLoopEnabled` option value.

Reviewer: ```suggestion
This option has no effect when [shading](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#shading) is enabled. In this case, focus always stays within the pop-up content, regardless of the `tabFocusLoopEnabled` value.
```
